### ISSUE TITLE
onDetect does not trigger in android 13

### ISSUE THREAD

**Issue by VictorWork0721** [juliansteenbakker/mobile_scanner#925]

**Comment by navaronbracke** [comment]
Are you still getting barcodes from the controller itself? The `onDetect` method is not the recommended way to listen to barcodes.

**Comment by VictorWork0721** [comment]
> Are you still getting barcodes from the controller itself? The `onDetect` method is not the recommended way to listen to barcodes.

How is the recommended way to listen to barcode?

**Comment by navaronbracke** [comment]
Listen directly to `MobileScannerController.barcodes` using Dart's Stream API:

https://github.com/juliansteenbakker/mobile_scanner/blob/master/lib/src/mobile_scanner_controller.dart#L88

```
final MobileScannerController = controller = MobileScannerController();

final Subscription<Object?> subscription = controller.listen((BarcodeCapture barcodes) {});

// ...
subscription.cancel();
```

**Comment by stephg** [comment]
> Listen directly to `MobileScannerController.barcodes` using Dart's Stream API:
> 
> https://github.com/juliansteenbakker/mobile_scanner/blob/master/lib/src/mobile_scanner_controller.dart#L88
> 
> ```
> final MobileScannerController = controller = MobileScannerController();
> 
> final Subscription<Object?> subscription = controller.listen((BarcodeCapture barcodes) {});
> 
> // ...
> subscription.cancel();
> ```

Looks like there is a typo:
```diff
- final Subscription<Object?> subscription = controller.listen((BarcodeCapture barcodes) {});
+ final Subscription<Object?> subscription = controller.barcodes.listen((BarcodeCapture barcodes) {});
```

If this is the recommended way of listening to the results, `onDetect` should not be required. Correct?
https://github.com/juliansteenbakker/mobile_scanner/blob/master/lib/src/mobile_scanner.dart#L78

**Comment by navaronbracke** [comment]
> > Listen directly to `MobileScannerController.barcodes` using Dart's Stream API:
> > https://github.com/juliansteenbakker/mobile_scanner/blob/master/lib/src/mobile_scanner_controller.dart#L88
> > ```
> > final MobileScannerController = controller = MobileScannerController();
> > 
> > final Subscription<Object?> subscription = controller.listen((BarcodeCapture barcodes) {});
> > 
> > // ...
> > subscription.cancel();
> > ```
> 
> Looks like there is a typo:
> 
> ```diff
> - final Subscription<Object?> subscription = controller.listen((BarcodeCapture barcodes) {});
> + final Subscription<Object?> subscription = controller.barcodes.listen((BarcodeCapture barcodes) {});
> ```
> 
> If this is the recommended way of listening to the results, `onDetect` should not be required. Correct? https://github.com/juliansteenbakker/mobile_scanner/blob/master/lib/src/mobile_scanner.dart#L78

Woops, yeah that's a typo. Indeed, `onDetect` should not have been a required parameter. It is a bit of a leftover from an older API signature on our end. I do have a PR open that fixes most issues with our public API, but that is blocked on a bug upstream in the Dart SDK.
Something broke on the web side of things.

For now you could pass an empty function to `onDetect`.
If you're feeling a bit bold, you could also switch to my WIP branch from that PR, but web support won't be working until https://github.com/dart-lang/sdk/issues/54534 is fixed.

See also: https://github.com/juliansteenbakker/mobile_scanner/pull/916

**Comment by VictorWork0721** [comment]
[https://github.com/juliansteenbakker/mobile_scanner/issues/698](url)
Done fixed. I just fixed with using 3.5.5 version and set the resolution based on the solution in this github issue. Thank you
